packages/kit/src/runtime/client/renderer.js

```diff
@@ -384,6 +384,16 @@ export class Renderer {
 					deep_linked.scrollIntoView();
 				} else {
 					scrollTo(0, 0);
+					if (document.body.scrollTop > 0) document.body.scrollTo(0, 0);
+					// In some cases it is needed scroll first root element of target that
+					// is not scrolled to the top
+					for (const target_child of this.target.childNodes) {
+						const target_child_element = target_child.firstChild?.parentElement;
```

Comment:
why do you get the parent of the child? wouldn't that just give you back the original element?

Reply:
target_child is the node and not the element itself. As element nodes must have at least one child in order to be an element-node (the node could also be just text) I implicitly check this by selecting the parent of the child. Also the node object does not have the same properties that an element object has. I'm happy to change the implementation if there is a better way 😄

Reply:
you can use [`this.target.children`](https://developer.mozilla.org/en-US/docs/Web/API/Element/children), which only includes elements